File: docs/sdk-documentation/getting-started/import-and-initialize/js/customEventLogger.js
const sdk = new absmartly.SDK({
  endpoint: "https://sandbox-api.absmartly.com/v1",
  apiKey: process.env.ABSMARTLY_API_KEY,
  environment: process.env.NODE_ENV,
  application: process.env.APPLICATION_NAME,
  eventLogger: (context, eventName, data) => {
    if (eventName == "error") {
      console.error(data);
    }
  },
});

File: docs/APIs-and-SDKs/SDK-Documentation/getting-started/custom-event-logger/liquid/customEventLogger.rb
event_logger = ->(context, event_name, data) {
  case event_name
  when 'exposure'
    Rails.logger.info "ABsmartly exposure: #{data[:name]}"
  when 'goal'
    Rails.logger.info "ABsmartly goal: #{data[:name]}"
  when 'error'
    Rails.logger.error "ABsmartly error: #{data}"
  when 'ready'
    Rails.logger.info "ABsmartly context ready"
  when 'refresh'
    Rails.logger.info "ABsmartly context refreshed"
  when 'publish'
    Rails.logger.info "ABsmartly events published"
  when 'close'
    Rails.logger.info "ABsmartly context closed"
  end
}

sdk = ABSmartly::SDK.new(
  endpoint: ENV['ABSMARTLY_ENDPOINT'],
  api_key: ENV['ABSMARTLY_API_KEY'],
  application: 'my-store',
  environment: Rails.env,
  event_logger: event_logger
)
